### DISCUSSION TITLE
What's the deal with this other similar project?

### DISCUSSION THREAD

**Discussion by Lewington-pitsos** [hlky/stable-diffusion-webui#686] [Q&A]
[this one here](https://github.com/AUTOMATIC1111/stable-diffusion-webui)

Are they building for some kind of different use case or something?

**Comment by Ainaemaet** [comment]
It's just different I think. It's a little frustrating as in some ways I prefer it over hlky and vice-versa.

**Comment by ReinerBforartists** [comment]
A software name is a brand. And should just exist once. Guess what happens when every photo editing software calls itself photoshop. This can quickly escalate in a legal way. This needs to be resolved as quick as possible. So question is who was first with this name?

Just as a reminder, the main repo of stable diffusion webui is still called stable diffusion. My belly still hurts here too ...

**Reply by mchaker** [reply]
I think things work differently on GitHub, where anyone can "fork" a project off the main project tree (and that is a good thing)

Information wants to be free

**Reply by ReinerBforartists** [reply]
It's not about the repo name, it's about the brand. Now we have two software called stable diffusion webui. And one is wrong ...

**Reply by Ainaemaet** [reply]
Doesn't the fact this is all open-source negate that fact? 
Anybody can build off of stable-diffusion and do with it what they will - same goes for the code @ hlky/stable-diffusion-webui as it's licensed under GNU Affero General Public License (an open source license). 

Also, I'm not an expert in copyright by any means but unless the name itself is protected in some fashion, I believe it would be fair game for anyone to use. After all, this _is_ all free and open-source software right?

**Reply by EliEron** [reply]
stable-diffusion-webui is not a "brand" it is an open source project. One built upon other projects. Which is also why it's GUI resembled the other stable-diffusion-webui, they were both based on the same starting script written by an anonymous user. So they are essentially both forks of the same project hence having the same name. That's also why the main repo is simply named stable-diffusion. It is a fork of the official stable-diffusion. Nothing more and nothing less.

Having multiple programs called "Photoshop" is legally problematic because Photoshop is a trademarked term, if it wasn't then there would be 0 legal issues with you naming your own photo editor photoshop. Photoshop isn't the best example of this since it's arguably been genericized as a term at this point but the point still stands. 

As an aside finding the starting date of a repo is pretty simple using the Github API. This repo was created at exactly 2022-08-24 10:43:48 UTC and the other repo was created 2022-08-22 14:05:26 UTC so the other repo was technically first. But again I don't agree that it matters at all who was first in this case. Forks sharing the same name is completely OK.

**Reply by ReinerBforartists** [reply]
Thanks for the thumbs down for pointing at and caring about the legal aspect of this repo friends :/

> Doesn't the fact this is all open-source negate that fact?

@Ainamemaet, why should being open source negate law? The code is open source, not the brand.

> stable-diffusion-webui is not a "brand" it is an open source project.

Sorry to disagree here, but it is exactly that, the name of the software. And as such it is the brand. The open source software Blender for example is also such a brand. Or Debian. Or Gimp. Or Krita. You cannot simply fork it, apply your changes, and then call it the same name. And stealing the customers from the original software. Which might not even be aware that they are not using the original from the original manufacturer. This is not how open source is meant to work. You can use the source code for your own software. But not the name. The brand is and remains legal property of the one who has invented it, which is in most cases the original software.

> so the other repo was technically first.

And so it deserves the name and the brand. Sad but true ...

What you seem to mix here is to fork an open source project to keep the source code in the public. But this is a completely different cup of tea. This project here extends Stable Diffusion by a graphical UI. And alters the code of somebody elses project while still using the same name.

**Reply by mchaker** [reply]
You seem to have a problem with a fundamental way that github works. Perhaps running your own website will satisfy your requirements?

**Reply by ReinerBforartists** [reply]
I don't have a problem with how Github works. It's more that you still refuse to understand the problem. I know what i talk about, after 25 years being a developer, that's all.

Anyways, i warned you, the rest is not my cup of tea. I will not support software pirates though.

**Reply by grmrgecko** [reply]
I get what you're trying to say, however stable-diffusion-webui isn't a brand. There is no registered trademarks.

Blender is a trademark: https://www.blender.org/about/logo/
Debian is a trademark: https://tsdr.uspto.gov/#caseNumber=75386376&caseType=SERIAL_NO&searchType=statusSearch
GIMP is a trademark: https://tsdr.uspto.gov/#caseNumber=79272850&caseType=SERIAL_NO&searchType=statusSearch
Krita is a trademark: https://krita.org/en/item/krita-trademark-policy/

When it comes to forking projects on github, there are many people who forks, writes improvements, and submit pull requests. I personally have done this at https://github.com/GRMrGecko/stable-diffusion-webui and there is no problem with this. Where you run into problems is if you were to take the code written and claim copyright on it. People fork dead projects all the time to revive them without contributing upstream, only cases where they have to change the name is if there is a trademark involved.

**Comment by ooofest** [comment]
It's all open-source and people can fork off whichever branch they see to create something unique.  I have been using this repo since the early point when SD became consumer-enabled, though admit that AUTOMATIC1111 has since raced ahead in terms of feature integration (e.g., early inpainting/outpainting capability), a comprehensive Settings page with lots of configurability, Progress indicator out of the box, etc.

They are both good, essentially, and you can have each - and more - installed side-by-side to try out.
